bugbug/tools/code_review.py

```diff
@@ -1116,15 +1119,23 @@ def find_similar_patch_comments(self, patch: Patch, limit: int):
 
         patch_set = PatchSet.from_string(patch.raw_diff)
 
-        yielded = 0
+        # We want to avoid returning the same comment multiple times. Thus, if
+        # a comment matches multiple hunks, we will only consider it once.
+        max_score_per_comment: dict = {}
         for patched_file in patch_set:
             if not patched_file.is_modified_file:
                 continue
 
             for hunk in patched_file:
                 for result in self.find_similar_hunk_comments(hunk):
-                    yield result
+                    if result is not None and (
+                        result.id not in max_score_per_comment
+                        or result.score > max_score_per_comment[result.id].score
+                    ):
+                        max_score_per_comment[result.id] = result
+
+        results_with_score = sorted(
+            (result.score, result) for result in max_score_per_comment.values()
+        )
 
-                    yielded += 1
-                    if yielded >= limit:
-                        return
+        return [e[1] for e in results_with_score[-limit:]]
```

Comment:
I think we should return the payload only:
```suggestion
        return [result.payload for result in sorted_results[-limit:]]
```

(the suggestion depends on https://github.com/mozilla/bugbug/pull/4406#discussion_r1711690040)